auth/src/main/java/software/amazon/awssdk/auth/AwsExecutionAttributes.java

```diff
@@ -40,6 +42,34 @@
      */
     public static final ExecutionAttribute<Region> AWS_REGION = new ExecutionAttribute<>("AwsRegion");
 
+    /**
+     * The signing name of the service to be using in SigV4 signing
+     */
+    public static final ExecutionAttribute<String> SERVICE_SIGNING_NAME = new ExecutionAttribute<>("ServiceSigningName");
+
+    /**
+     * Option to specify whether to do double url encoding during signing. If not set, the default value is True
+     */
+    public static final ExecutionAttribute<Boolean> SIGNER_DOUBLE_URL_ENCODE = new ExecutionAttribute<>("DoubleUrlEncode");
+
+    /**
+     * Option to specify the modeled class with all the parameters required for the signing. Aws4SignerParams is the base class
+     * for parameters requires for AWS4Signing. Implementations can take Aws4SignerParams class or sub class as input.
+     *
+     * You need to specify the correct sub class based on the Signer you are using. For example, If you use S3 AwsS3V4Signer,
+     * the value you set for this attribute should be of "AwsS3V4SignerParams" type
+     *
+     * Class that contains all parameters required for SDK signer implementations
+     */
+    public static final ExecutionAttribute<Aws4SignerParams> AWS_SIGNER_PARAMS =
+        new ExecutionAttribute<>("Aws4SignerParams");
```

Comment:
What's the goal of this attribute?

Reply:
I added it so that customers don't need to set each individual attribute seperately in execution attributes. But as we have the overloaded sign method that takes the modeled input, I don't think this is needed anymore. Will remove